File: be/src/main/java/com/example/mingle/domain/chat/archive/dto/ArchiveItemResponse.java
package com.example.mingle.domain.chat.archive.dto;

import com.example.mingle.domain.chat.archive.entity.ArchiveItem;
import com.example.mingle.domain.chat.archive.entity.ArchiveTag;

import java.time.LocalDateTime;
import java.util.List;

public record ArchiveItemResponse (
        Long id,
        String fileUrl,
        String fileName,
        String thumbnailUrl,
        List<String> tags,
        String uploaderNickname,

        // 업로드 일시
        LocalDateTime createdAt
) {
    public static ArchiveItemResponse from(ArchiveItem item) {
        return new ArchiveItemResponse(
                item.getId(),
                item.getFileUrl(),
                item.getOriginalFilename(),
                item.getThumbnailUrl(),
                item.getTags().stream().map(ArchiveTag::getName).toList(),
                item.getUploader().getNickname(),
                item.getCreatedAt()
        );
    }
}

File: fe/src/features/chat/archive/types/ArchiveItemResponse.ts
// 서버의 ArchiveItemResponse DTO에 대응하는 프론트 타입

export interface ArchiveItemResponse {
  id: number;
  fileUrl: string;
  fileName: string;
  thumbnailUrl: string | null;
  tags: string[];
  uploaderNickname: string;
  createdAt: string; // ISO 8601 형식 문자열
}
